Missing 'require (or use) Config' in t/opbasic/arith.t

t-a-k opened the pull request:
I found that `t/opbasic/arith.t` refers `%Config` without loading `Config.pm`, so that some tests were never exercised.

t-a-k committed:
t/opbasic/arith.t: load Config.pm

arith.t used to refer %Config without loading Config.pm, so that
some tests were never exercised.

Leont commented on t/opbasic/arith.t:
```diff
@@ -429,16 +429,20 @@ $a = (97656250000000000 % $1);
 $b = (97656250000000000 % "$1");
 print "not "x($a ne $b), "ok ", $T++, qq ' - something % \$1 vs "\$1"\n';
 
+unless (eval { require Config }) {
```
Why not just «use Config;»?

tonycoz replied:
`t/opbasic/*` is run by minitest and shouldn't depend on Config loading successfully.

It shouldn't be loading test.pl either (which is partly my mistake).

Also, the change is wrong, since it doesn't import Config.

t-a-k replied:
> Why not just «use Config;»?

My change uses `eval { require ... }` because `arith.t` already had it in `if ($^O eq 'VMS')` block, and I first started this change with moving this `eval` out of the if-block.
And I didn't changed `require` to `use` because I think `arith.t` is a basic test and should be run (at least partially) even if `Config` couldn't be loaded successfully.

I think it might be better to split Config-dependent tests to a separate file, for example `t/op/arith2.t`.

t-a-k committed:
Split Config-dependent tests in t/opbasic/arith.t to t/op/arith2.t

Some tests in t/opbasic/arith.t used to depend on the perl configuration,
but t/opbasic/* is so basic tests that it should not depend on Config.pm.
Now these tests are splitted into separate test file under t/op/.

t-a-k commented:
I fixed this PR to propose splitting tests that depend on Config into a separate file, `t/op/arith2.t`.

jkeenan commented:
> I fixed this PR to propose splitting tests that depend on Config into a separate file, `t/op/arith2.t`.

However, you have merge conflicts.  Can you correct?

Thank you very much.
Jim Keenan

t-a-k commented:
> However, you have merge conflicts. Can you correct?

Okay, I will try to correct it shortly.

demerphq commented:
Our of curiosity is this related to

4de2690c76c1d4569e6e698f0cf54b7ec8e183b9

Didnt I fix this already?


On Mon, 15 Feb 2021 at 12:01, TAKAI Kousuke <notifications@github.com>
wrote:

> However, you have merge conflicts. Can you correct?
>
> Okay, I will try to correct it shortly.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Perl/perl5/pull/18519#issuecomment-779143290>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAZ5R7SLORZBG7VZBK2ORDS7D5G5ANCNFSM4W5EUX7A>
> .
>


-- 
perl -Mre=debug -e "/just|another|perl|hacker/"

t-a-k committed:
Merge branch 'blead' into fix-arith.t-config

t-a-k commented:
> Our of curiosity is this related to
[4de2690](https://github.com/Perl/perl5/commit/4de2690c76c1d4569e6e698f0cf54b7ec8e183b9)
Didnt I fix this already?

I'm afraid that the commit 4de2690 didn't fix this issue, because it still load Config.pm on VMS only. On the other architectures `%Config::Config` is empty and some tests would be skipped unintentionally.

In addition to that, `t/opbasic/*` shouldn't depend on Config (as @tonycoz pointed out above), therefore I finally splitted the tests that depend on Config into a separate test file in `t/op/`.

demerphq commented:
On Mon, 15 Feb 2021 at 19:26, TAKAI Kousuke <notifications@github.com>
wrote:

> Our of curiosity is this related to
> 4de2690
> <https://github.com/Perl/perl5/commit/4de2690c76c1d4569e6e698f0cf54b7ec8e183b9>
> Didnt I fix this already?
>
> I'm afraid that the commit 4de2690
> <https://github.com/Perl/perl5/commit/4de2690c76c1d4569e6e698f0cf54b7ec8e183b9>
> didn't fix this issue, because it still load Config.pm on VMS only. On the
> other architectures %Config::Config is empty and some tests would be
> skipped unintentionally.
>
> In addition to that, t/opbasic/* shouldn't depend on Config (as @tonycoz
> <https://github.com/tonycoz> pointed out above), therefore I finally
> splitted the tests that depend on Config into a separate test file in
> t/op/.
>
I was under the impression that it only mattered on VMS. But fair enough.


Yves
-- 
perl -Mre=debug -e "/just|another|perl|hacker/"
